Tella/Scenes/Resources/ViewModel/ResourcesViewModel.swift

```diff
@@ -0,0 +1,218 @@
+                        }
+                        self.isLoadingList = false
+                    }
+
+                },
+                receiveValue: { resources in
+                    self.isLoadingList = false
+                    let downloadedIds = Set(self.downloadedResources.map { $0.externalId })
+                    let newResources = resources.filter { !downloadedIds.contains($0.id) }
+                    self.availableResources.append(contentsOf: newResources)
+
+                }
+            ).store(in: &cancellables)
+    }
+    
+    func fetchAvailableResources() -> AnyPublisher<
+        [AvailableResourcesList], APIError
+    > {
+        // Logic to fetch available resources
+        let resourceRepository = ResourceRepository()
+        let publishers = self.servers.map { server in
+            resourceRepository.getResourcesByProject(server: server)
+                .map { response in
+                    response.map { resource -> AvailableResourcesList in
+                        AvailableResourcesList(
+                            id: resource.id,
+                            resourceCard: ResourceCardViewModel(
+                                title: resource.title,
+                                serverName: server.name ?? "",
+                                type: .save,
+                                action: {self.downloadResource(serverName: server.name ?? "", resource: resource)}
+                            ),
+                            fileName: resource.fileName,
+                            size: resource.size,
+                            createdAt: resource.createdAt
+                        )
+                    }
+                }
+        }
+
+        return Publishers.MergeMany(publishers)
+            .collect()
+            .map { $0.flatMap { $0 } } 
+            .eraseToAnyPublisher()
+    }
+
+    func downloadResource(serverName: String, resource: Resource) {
+        toggleIsLoadingResource(id: resource.id)
+        guard let selectedServer = servers.first(where: { $0.name == serverName }) else {
+            return
+        }
+
+        fetchResourceFromServer(server: selectedServer, resource: resource) {
+            result in
+            switch result {
+            case .success(let data):
+                self.saveToVault(data: data, resource: resource, serverId: selectedServer.id!)
+            case .failure(let error):
+                switch error {
+                case .noInternetConnection:
+                    debugLog("Error downloading file: \(error)")
+                    Toast.displayToast(message: error.errorDescription ?? error.localizedDescription)
+                default:
+                    debugLog("Error downloading file: \(error)")
+                    Toast.displayToast(message: error.localizedDescription)
+                }
+                self.toggleIsLoadingResource(id: resource.id)
+            }
+        }
+    }
+    
+
+    func fetchResourceFromServer(
```

Comment:
In my opinion, I see that it will add more code, if you want to make it more readeable and improve it, you can create function to handle errors and another one to handle sucess result

maybe if we can find a solution to convert the Publisher to async/await -> throws, but we still in this case need to add Task, do catch